### PR TITLE
Pass Mutation Name as Endpoint for GraphQL

### PR DESCRIPTION
**Pull request by richherrera**
### Summary of changes

- For requests that go through GraphQL, we need to send the name of the mutation as the endpoint for our latency events

### Checklist

- [ ] Added a changelog entry

### Authors

- @richherrera

### CODE REVIEW

File: Sources/BraintreeCore/BTHTTP.swift [hunk 1]
```diff
@@ -390,7 +390,14 @@ class BTHTTP: NSObject, URLSessionTaskDelegate {
         metrics.transactionMetrics.forEach { transaction in
             if let startDate = transaction.fetchStartDate,
                let endDate = transaction.responseEndDate,
-               let path = transaction.request.url?.path {
+               var path = transaction.request.url?.path {
+                
+                if path.contains("graphql"),
```
Comment (jaxdesmarais) [inline Sources/BraintreeCore/BTHTTP.swift:395]: Are we able to add a unit test for this?

Comment (richherrera) [reply Sources/BraintreeCore/BTHTTP.swift:395]: Sure, added: [df95e39](https://github.com/braintree/braintree_ios/pull/1349/commits/df95e39eb534c777de12ca45a50e2c8e3fd5d7fe)

File: UnitTests/BraintreeCoreTests/BTHTTP_Tests.swift [hunk 2]
```diff
@@ -749,6 +749,36 @@ final class BTHTTP_Tests: XCTestCase {
...
+        originalRequest.httpBody = """
+            {
+                "operationName": "TestMutation"
+            }
+            """.data(using: .utf8)
+        let task = testURLSession.dataTask(with: originalRequest)
+
+        let transactionMetrics = MockURLSessionTaskTransactionMetrics()
+        transactionMetrics.mockConnectStartDate = Date()
+        transactionMetrics.mockFetchStartDate = Date()
```
Comment (agedd) [inline UnitTests/BraintreeCoreTests/BTHTTP_Tests.swift:767]: just curious, how are these two values different? 👀

**Review by agedd — APPROVED** [review]
lgtm! 🚀 jw what brought about introducing this change

**Comment by richherrera** [comment]
> lgtm! 🚀 jw what brought about introducing this change

Thanks for asking. Yes, this change is part of something that was pending in another PR. Currently, we're sending the endpoint of all requests to measure latency. Since the GraphQL requests are the same, the suggestion was to send the name of the mutation to identify it. I'll share some PRs with you in case you want to dive deeper into this.

Suggestion: https://github.com/braintree/braintree_ios/pull/1321#discussion_r1617762574
Android: [78883ff](https://github.com/braintree/braintree_android/pull/1032/commits/78883ffe6df7c9d0a5b766f600ea21305e4effd7)

File: UnitTests/BraintreeCoreTests/BTHTTP_Tests.swift (hunk 2 above)
Comment (richherrera) [reply UnitTests/BraintreeCoreTests/BTHTTP_Tests.swift:767]: > FetchStartDate: The time when the task started fetching the resource, from the server or locally.
> ConnectStartDate: The time immediately before the task started establishing a TCP connection to the server.

Apple has an amazing graphic in their documentation that would explain it better. [Apple doc reference](https://developer.apple.com/documentation/foundation/urlsessiontasktransactionmetrics)

**Review by sshropshire — APPROVED** [review]

**Review by scannillo — APPROVED** [review]
Nice unit test 👏

**Pull request merged**